modules/gapi/include/opencv2/gapi/infer/ie.hpp

```diff
@@ -117,75 +136,171 @@ template<typename Net> class Params {
+    outputs.
+
+    @param layer_names array that contains names of output layers.
+    @return reference to object of class Params.
+    */
+    Params<Net>& cfgOutputLayers(const typename PortCfg<Net>::Out &layer_names) {
         desc.output_names.clear();
-        desc.output_names.reserve(ll.size());
-        std::copy(ll.begin(), ll.end(),
+        desc.output_names.reserve(layer_names.size());
+        std::copy(layer_names.begin(), layer_names.end(),
                   std::back_inserter(desc.output_names));
         return *this;
     }
 
+    /** @brief Sets constant input.
+
+    The function is used to set constant input. This input has to be
+    a prepared tensor since preprocessing is disabled for this case. You should
+    provide data and name of CNN layer which will be associated with it.
+
+    @param layer_name name of CNN layer.
+    @param data cv::Mat that contains data which will be associated with CNN layer.
+    @param hint type of input (IMAGE or TENSOR).
+    @return reference to object of class Params.
+    */
     Params<Net>& constInput(const std::string &layer_name,
                             const cv::Mat &data,
                             TraitAs hint = TraitAs::TENSOR) {
         desc.const_inputs[layer_name] = {data, hint};
         return *this;
     }
 
-    Params& pluginConfig(IEConfig&& cfg) {
-        desc.config = std::move(cfg);
+    /** @brief Sets IE config.
+
+    The function is used to set configuration for device.
+
+    @param cfg map of pairs: (config parameter name, config parameter value).
+    @return reference to object of class Params.
+    */
+       Params& pluginConfig(const IEConfig& cfg) {
+        desc.config = cfg;
         return *this;
     }
 
-    Params& pluginConfig(const IEConfig& cfg) {
-        desc.config = cfg;
+    /** @overload
+    Function with rvalue parameter.
+
+    @param cfg rvalue map of pairs: (config parameter name, config parameter value).
+    @return reference to object of class Params.
+    */
+    Params& pluginConfig(IEConfig&& cfg) {
+        desc.config = std::move(cfg);
         return *this;
     }
 
+    /** @brief Sets count of asynchronous inference requests.
+
+    @param nireq count of inference asynchronous requests.
+    @return reference to object of class Params.
+    */
     Params& cfgNumRequests(size_t nireq) {
         GAPI_Assert(nireq > 0 && "Number of infer requests must be greater than zero!");
         desc.nireq = nireq;
         return *this;
     }
 
-    Params<Net>& cfgInputReshape(std::map<std::string, std::vector<std::size_t>>&& reshape_table) {
-        desc.reshape_table = std::move(reshape_table);
-        return *this;
-    }
+    /** @brief Sets new input shapes for the CNN.
 
+    The function is used to set new input shapes for network. CNN will be reshaped by
+    new shapes.
+
+    @param reshape_table map of pairs: name of corresponding data and its dimension.
+    @return reference to object of class Params.
```

Comment:
Please explain more how does it behave.

Should all the known input layers be specified here or not? What happens when only particular inputs are reshaped? If the input is reshaped and the input image is of other size, what happens in this case?

Reply:
Changed message about reshape.